obs/error.go

```diff
@@ -0,0 +1,22 @@
+package obs
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+type Status struct {
+	XMLName xml.Name `json:"status" xml:"status"`
+	Code    string   `json:"code" xml:"code,attr"`
+	Summary string   `json:"summary" xml:"summary"`
+}
+
+type APIError struct {
+	HTTPStatusCode int
+	XMLStatusCode  string
+	Message        string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("HTTP status %d: %s (%s)", e.HTTPStatusCode, e.XMLStatusCode, e.Message)
```

Comment:
Does it make sense to have a different error message if HTTP status code is 0?

Reply:
we generally have the message as `unable to generate the request`.